CMakeLists.txt

```diff
@@ -87,14 +88,53 @@ include(cmake/secp256k1.cmake)
 include(CheckStdFilesystem)
 check_std_filesystem()
 
-set(THREADS_PREFER_PTHREAD_FLAG ON)
-find_package(Threads REQUIRED)
+find_package(PkgConfig)
+if(BUILD_DAEMON)
+  set(THREADS_PREFER_PTHREAD_FLAG ON)
+  find_package(Threads REQUIRED)
+  if(NOT MINGW AND NOT CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
+    target_compile_definitions(Threads::Threads INTERFACE
+      $<$<COMPILE_FEATURES:cxx_thread_local>:HAVE_THREAD_LOCAL>
+    )
+  endif()
+
+  if(APPLE AND BREW_COMMAND)
+    execute_process(
+      COMMAND ${BREW_COMMAND} --prefix boost
+      OUTPUT_VARIABLE BOOST_ROOT
+      ERROR_QUIET
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+  endif()
+  set(Boost_NO_BOOST_CMAKE ON)
+  # Find Boost headers only.
+  find_package(Boost 1.64.0 REQUIRED)
+  mark_as_advanced(Boost_INCLUDE_DIR)
+  target_compile_definitions(Boost::boost INTERFACE
+    $<$<CONFIG:Debug>:BOOST_MULTI_INDEX_ENABLE_SAFE_MODE>
+  )
+  if(CMAKE_VERSION VERSION_LESS 3.15)
```

Comment:
See other comment, but I think this is reasonable assuming the behavior is actually the same. Needs a comment though please.